Packages/Components/Sources/CachedAsyncNFTImage.swift

```diff
@@ -0,0 +1,161 @@
+
+import Foundation
+import SwiftUI
+
+public enum AsyncNFTImagePhase {
+    case empty
+    case success(Image)
+    case failure(Error)
+
+    public var image: Image? {
+        if case let .success(img) = self { return img }
+        return nil
+    }
+
+    public var error: Error? {
+        if case let .failure(err) = self { return err }
+        return nil
+    }
+}
+
+public struct CachedAsyncNFTImage<Content>: View where Content: View {
```

Comment:
```suggestion
public struct CachedAsyncNFTImage<Content>: View where Content: View {
```

should be generic, not nft specific